AlphaTechnic/users/models.py

```diff
@@ -0,0 +1,34 @@
+from django.db import models
+from django.contrib.auth.models import User
+from PIL import Image
+
+
+class Profile(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    image = models.ImageField(default='users/profile_pic/default.png', upload_to="users/profile_pic")
+
+    def __str__(self):
+        return f'{self.user.username} Profile'
+
+    @property
+    def followers(self):
+        return Follow.objects.filter(follow_user=self.user).count()
+
+    @property
+    def following(self):
+        return Follow.objects.filter(user=self.user).count()
+
+    def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
+        super().save()
+
+        img = Image.open(self.image.path)
+        if img.height > 300 or img.width > 300:
+            size = (300, 300)
+            img.thumbnail(size)
+            img.save(self.image.path)
+
+
+class Follow(models.Model):
+    user = models.ForeignKey(User, related_name='user', on_delete=models.CASCADE)
+    follow_user = models.ForeignKey(User, related_name='follow_user', on_delete=models.CASCADE)

```

Comment:
필드명이 조금 헷갈리는데 `user`는 본인이 팔로윙하는 유저를 말하고, `follow_user`는 본인을 팔로우하는 유저를 말하나요?

스터디 때 말씀 드린 것 처럼 `related_name`의 네이밍을 조금 수정하는 것이 어떨까요??

```suggestion
    user = models.ForeignKey(User, related_name='followings', on_delete=models.CASCADE)
    follow_user = models.ForeignKey(User, related_name='followers', on_delete=models.CASCADE)

```

Reply:
오 넵! 이부분 반드시 수정하겠습니다 ㅎㅎ

Reply:
제가 이걸 왜 이렇게 만들었을까 보다가 생각이 났는데요.
제가 이걸 맨처음에 followees, followers 이렇게 했다가 헷갈릴 것 같아서 바꾼건데, 왜그랬냐면 주어(user)랑 목적어(user가follow하는애들)가 분명했으면 했거든요~
"해당하는 user를 찾아서 그 user가 follow하고 있는 애들을 알려줘라" 이런식으로 view에서 만질려고요!
followee랑 follower 이렇게 하면 주어, 목적어가 분명해지지 않아서 이런 네이밍을 했던건데 피드백 주실수 있으실까용??!

Reply:
그러셨군요~ 네이밍을 말씀 드리는 것 자체가 저도 썩 편하지가 않네요. (뭔가 쪼잔한 그런 느낌이....ㅎㅎ)
생각하실 때 저게 더 편하시면 저렇게 하셔도 무방합니당!!😉